pkg/apis/pipeline/v1/container_validation.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"knative.dev/pkg/apis"
+)
+
+// Validate ensures that a supplied Ref field is populated
+// correctly. No errors are returned for a nil Ref.
+func (ref *Ref) Validate(ctx context.Context) (errs *apis.FieldError) {
```

Comment:
this is quite duplicate for pipelinere, taskref and stepaction's ref. Not sure if we have plans to unify them?

Reply:
Could you please do this cleanup then we can merge that ahead of this PR? 🙏🏾

Reply:
I think we cannot do it now? I was thinking about merging pipelineref, taskref and stepaction's ref into 1 ref. so they can share the same validation. But they have different fields, e,g, the taskref has fields (kind) for customtask

Reply:
ah I see, all good, thought you meant that it's possible now

this issue is relevant to this discussion: https://github.com/tektoncd/pipeline/issues/5138

maybe we can explore that as we work towards v2

Reply:
Yes, exactly! Now we have stepaction ref, the the similar validation now happens in 3 places. I'm hoping we can merge them in the future.